### PR TITLE
[IZPACK-1751] Fixes 'inconsistent archive'

### PR DESCRIPTION
**Pull request by reinhapa**
- Properly closing the JarInputStream after installer creation
- Using diamond operators
- Use of NIO instead old File

### CODE REVIEW

File: izpack-compiler/src/main/java/com/izforge/izpack/compiler/packager/impl/PackagerBase.java [hunk 1]
```diff
@@ -332,48 +349,48 @@ public Map<String, List<DynamicVariable>> getDynamicVariables()
...
+    public final void addInstallerRequirements(List<InstallerRequirement> conditions)
     {
         this.installerRequirements = conditions;
     }
 
     @Override
-    public void createInstaller() throws Exception
+    public final void createInstaller() throws Exception
     {
+        System.err.println("******* start createInstaller");
```
Comment (akuhtz) [inline izpack-compiler/src/main/java/com/izforge/izpack/compiler/packager/impl/PackagerBase.java:366]: Why on System.err and not use logger? Same on line 378.

**Review by akuhtz — CHANGES_REQUESTED** [review]

File: izpack-compiler/src/main/java/com/izforge/izpack/compiler/container/provider/JarOutputStreamProvider.java [hunk 2]
```diff
@@ -37,35 +38,30 @@
...
+            else
             {
                 jarOutputStream.setLevel(Deflater.BEST_COMPRESSION);
             }
+            return jarOutputStream;
         }
         catch (IOException e)
         {
-            IOUtils.closeQuietly(fileOutputStream);
+            throw new IllegalStateException((e));
```
[the remark this answers is not in the document]
Comment (reinhapa) [reply izpack-compiler/src/main/java/com/izforge/izpack/compiler/container/provider/JarOutputStreamProvider.java]: Missed that, thanks...

File: izpack-compiler/src/main/java/com/izforge/izpack/compiler/packager/impl/PackagerBase.java (hunk 1 above)
Comment (reinhapa) [reply izpack-compiler/src/main/java/com/izforge/izpack/compiler/packager/impl/PackagerBase.java]: Forgot to remove after locating the bug...

**Review by akuhtz — APPROVED** [review]
LGTM

**Pull request merged**